src/App.tsx

```diff
@@ -1,61 +1,193 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
 import './App.scss';
+import 'bulma/css/bulma.css';
+
+import { TodoList } from './components/TodoList';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todosInitial: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+let nextTodoId = Math.max(...todosInitial.map(todo => todo.id)) + 1;
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosInitial);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [isTitleErrorShown, setIsTitleErrorShown] = useState(false);
+  const [isUserIdErrorShown, setIsUserIdErrorShown] = useState(false);
+
+  const validateFields = (): boolean => {
+    let isFieldsValid = true;
+
+    if (!title) {
+      isFieldsValid = false;
+      setIsTitleErrorShown(true);
+    }
+
+    if (!userId) {
+      isFieldsValid = false;
+      setIsUserIdErrorShown(true);
+    }
+
+    return isFieldsValid;
+  };
+
+  const addNewTodo = (): void => {
+    const newTodo: Todo = {
+      id: nextTodoId,
+      userId,
+      title,
+      completed: false,
+      user: getUserById(userId),
+    };
+
+    setTodos((prevTodos) => [
+      ...prevTodos,
+      newTodo,
+    ]);
+
+    nextTodoId += 1;
+  };
+
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    const isEachFieldValid = validateFields();
```

Comment:
```suggestion
    const isFieldsValid = validateFields();
```

Reply:
Sergii recommended an [article ](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig#:~:text=Array.prototype.every-,isEachUserLoggedIn,-%F0%9F%A5%B0)which proposed a naming convention specifically for Boolean variables that I think is better to follow :)